src/main/java/org/terasology/launcher/packages/LocalStorage.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.packages;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Provides game packages from the local filesystem.
+ */
+public class LocalStorage implements Storage {
+
+    private static final Logger logger = LoggerFactory.getLogger(LocalStorage.class);
+
+    private static final String VERSIONS_CACHE_FILENAME = "versions.cache";
+
+    private final Path versionsCache;
+    private final Map<GamePackageType, List<Integer>> cachedVersions;
+
+    LocalStorage(Path localDir) {
+        versionsCache = localDir.resolve(VERSIONS_CACHE_FILENAME);
+
+        final Map<GamePackageType, List<Integer>> cache = loadCache();
+        if (!cache.isEmpty()) {
+            cachedVersions = cache;
+        } else {
+            cachedVersions = new EnumMap<>(GamePackageType.class);
+            for (GamePackageType pkgType : GamePackageType.values()) {
+                cachedVersions.put(pkgType, Collections.emptyList());
+            }
+        }
+    }
+
+    void updateCache(GamePackageType pkgType, List<Integer> versions) {
```

Comment:
More access modifier. The IDE might sometime offer premature advice about what access level is needed, but it is rare to use anything beyond `public` or `private` (let alone package access level on purpose)